pkg/odo/cli/utils/utils.go

```diff
@@ -36,3 +42,63 @@ func VisitCommands(cmd *cobra.Command, f func(*cobra.Command)) {
 		VisitCommands(child, f)
 	}
 }
+
+// LocalConfigInfo is a struct that will contain the LocalConfig as well as SourcePath
+// the reasoning behind SourcePath is due to the fact that we must correct what the path is
+// on different platforms (Linux, Windows, etc.) as well as get the absolute path of the component.
+type LocalConfigInfo struct {
+	LocalConfig *config.LocalConfigInfo
+	SourcePath  string
+}
+
+// RetrieveLocalConfigInfo retrieves the local configuration
+func RetrieveLocalConfigInfo(componentContext string) (configInfo LocalConfigInfo, err error) {
+
+	conf, err := config.NewLocalConfigInfo(componentContext, false)
+	if err != nil {
+		return LocalConfigInfo{}, errors.Wrap(err, "failed to fetch component config")
+	}
+
+	sourcePath, err := correctSourcePath(conf)
+	if err != nil {
+		return LocalConfigInfo{}, errors.Wrap(err, "unable to validate source path")
+	}
+
+	return LocalConfigInfo{
+		LocalConfig: conf,
+		SourcePath:  sourcePath}, nil
+}
+
+// correctSourcePath corrects the current sourcePath depending on local or binary configuration
+func correctSourcePath(localConfig *config.LocalConfigInfo) (path string, err error) {
+
+	cmpName := localConfig.GetName()
+	sourceType := localConfig.GetSourceType()
+	sourcePath := localConfig.GetSourceLocation()
+
+	if sourceType == config.BINARY || sourceType == config.LOCAL {
```

Comment:
Can the lines 79 to 88 be moved instead to pkg/config/config.go#GetSourceLocation instead ?

Reply:
I don't think so. The only reasoning is that if we add it to GetSourceLocation:

```sh
// GetSourceLocation returns the sourcelocation, returns default if nil
func (lc *LocalConfig) GetSourceLocation() string {
  if lc.componentSettings.SourceLocation == nil {
    return ""
  }
  return *lc.componentSettings.SourceLocation
}
```

It would impact the general functionality of it. It would no longer "get" the value, but the function would interpolate something different...